File: 0442_find_all_duplicates_in_an_array/Solution.js
var findDuplicates = function (nums) {
    let duplicatedNums = [];
    let freq = new Array(nums.length + 1);
    freq.fill(0);

    nums.forEach((n) => {
        freq[n]++;
    });

    freq.forEach((n, i) => {
        if (n > 1) duplicatedNums.push(i);
    });

    return duplicatedNums;
};

File: 0442_find_all_duplicates_in_an_array/Solution.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> duplicatesNums = new ArrayList<>();
        int[] temp = new int[nums.length + 1];

        for (int n : nums) {
            temp[n]++;
        }

        int i = 0;
        for (int n : temp) {
            if (n > 1) duplicatesNums.add(i);
            i++;
        }

        return duplicatesNums;
    }
}
